public/js/core.js

```diff
@@ -37,12 +37,14 @@ app.controller('mainController', function ($scope, $state, Todo) {
 
     $scope.createTodo = function () {
         $scope.loading = true;
-
-        if ($scope.newTodo.text !== '') {
+        if ($scope.newTodo.text !== "") {
```

Comment:
A better way to check for the empty string is 
```javascript
 if (!$scope.newTodo.text) {
...
```
Here you will check your variable also for `undefined` and `null`